core/src/helpers.rs

```diff
@@ -0,0 +1,27 @@
+use eyre::Result;
+use starknet::{
+    core::types::FieldElement,
+    providers::jsonrpc::models::{BlockId as StarknetBlockId, BlockTag},
+};
+extern crate hex;
+// use rustc_hex::ToHex;
+
+use reth_primitives::rpc::{BlockId, BlockNumber};
+
+pub fn ethers_block_id_to_starknet_block_id(block: BlockId) -> Result<StarknetBlockId> {
+    match block {
+        BlockId::Hash(hash) => {
+            let address_hex = hex::encode(hash);
+            let address_felt = FieldElement::from_hex_be(&address_hex).unwrap();
```

Comment:
We should "almost" never use `unwrap`, unless we're certain that the code will not actually error out:)
Generally, if one provides as a comment above the line of code the reason why their code cannot error, i'm ok with unwrap,
otherwise, let's always use error management using `?`

Reply:
I'll create a PR to handle errors I guess!:)

Reply:
Ok!